website/docs/attached-class.md

```diff
@@ -0,0 +1,75 @@
+---
+id: attached-class
+title: T.attached_class
+---
+
+`T.attached_class` can be used to refer to the type of instances from a
+singleton-context, in a way that is friendly to inheritance.
+
+```ruby
+class AbstractData
+  extend T::Sig
+
+  sig {params(id: Integer).returns(T.attached_class)}
+  def self.load(id)
+    new
+  end
+end
+
+class Article < AbstractData; end
+class User < AbstractData; end
+class Administrator < User; end
+
+T.reveal_type(Article.load(10)) # Article
+T.reveal_type(Administrator.load(20)) # Administrator
+```
+
+# Typing Factory Methods
+
+Consider the following example of a class hierarchy with a factory method:
+
+```ruby
+class Parent
+  extend T::Sig
+
+  sig {returns(Parent)}
+  def self.make
+    new
+  end
+end
+
+class Child < Parent; end
+
+T.reveal_type(Parent.make) # Parent
+T.reveal_type(Child.make) # Parent
+```
+
+`Parent` defines a single factory method `self.make`, which will create an
+instance of the `Parent` class. When this method is called off of the `Parent`
+class, it returns a new instance of type `Parent`. When it is called on `Child`
+it creates an instance of `Child` but the signature of `Parent.make` will cause
+the return value to be up-cast to `Parent`. This is fine if it's convenient to
+use constructed `Child` values as though they were `Parent`, but becomes
+problematic if we need to call methods that are unique to `Child` instances.
```

Comment:
You might be able to motivate most of this paragraph by defining a method on the child, calling `Child.make.that_method` and then showing that Sorbet would report an error.

i.e., show what you mean, rather than say what you mean.